### PR TITLE
Consolidate node documentation

### PR DESCRIPTION
**Pull request by janewang**
* Move all node related docs into the same dir
* Fix links
* Update sidebar
* WIP: Need more info for RPC and Validator node

### CODE REVIEW

**Comment by janewang** [comment]
Screen Shot 2021-06-07 at 1 06 34 PM
CI test-link is complaining about broken link on master, not the feature branch ¯\_(ツ)_/¯.

File: docs/develop/node/community/node-community-updates.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+id: node-community-updates
+title: Node Community Updates
+sidebar_label: Node Community Updates
+---
+
+## NEAR Validator Community Channels
+
+NEAR Protocol will communicate with validators using these channels:
+
+1. New releases, technical support and bug tracking in the Github repository of [`nearcore`](https://github.com/near/nearcore/issues) and Github project [`Node Experience`](https://github.com/orgs/near/projects/18).
```
Comment (bowenwang1996) [inline docs/develop/node/community/node-community-updates.md:11]: let's separate new releases from technical support and bug tracking to be more explicit

File: docs/develop/node/intro/types-of-node.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+---
+id: types-of-node
+title: Types of Node
+sidebar_label: Types of Node
+---
+
+You can run three different types of node - Validator Node, RPC Node, and Archival Node.
+
+## Validator Node
+Validator Nodes are the operators of the NEAR blockchain and are essential to the health of the network. Validator nodes receive new states and blocks while participating in block validation, and verify blocks and states.
```
Comment (bowenwang1996) [inline docs/develop/node/intro/types-of-node.md:10]: Note that nonvalidator nodes also verify blocks. I think it is better to say something like "Validator nodes participate in the consensus and produce blocks and/or chunks".

File: docs/develop/node/intro/types-of-node.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+## RPC Node
+RPC Nodes are RPC service providers for that provide public RPC endpoints for developers to use. The NEAR Inc currently maintains a public RPC endpoint `http://rpc.mainnet.near.org/` that is free for everyone to use. However, any participants are encouraged to run their own RPC node and offer RPC services through a grant from the NEAR Foundation. For more information, please reach out in Validator channels on Telegram or Discord.
```
Comment (bowenwang1996) [inline docs/develop/node/intro/types-of-node.md:13]: The rpc endpoint is technically provided by NEAR Foundation I believe. cc @eriktrautman @ilblackdragon

File: docs/develop/node/maintenance/maintenance.md [hunk 4]
```diff
@@ -42,9 +30,13 @@ Once `MainNet: Restricted` is live, planned updates to `TestNet` and `MainNet` w
 
 ## Emergency Updates
 
+We may issue a `[CODE_YELLOW_BETANET]` or `[CODE_YELLOW_TESTNET]` if the network is suffering minor issues, or a new software release introduces incompatibilities and requires additional testing.
```
Comment (bowenwang1996) [inline docs/develop/node/maintenance/maintenance.md:33]: Don't think we do this for betanet. Probably better to use mainnet as an example

File: docs/develop/node/maintenance/maintenance.md [hunk 5]
```diff
@@ -42,9 +30,13 @@ Once `MainNet: Restricted` is live, planned updates to `TestNet` and `MainNet` w
...
+
 NEAR Protocol team will use the tag `[CODE_RED_BETANET]` or `[CODE_RED_TESTNET]` in the Validator Announcement channel on [Discord](https://discord.gg/xsrHaCb), followed by email instructions for coordination. Some updates may follow a confidential process, as explained on [nearcore/SECURITY.md](https://github.com/near/nearcore/blob/master/SECURITY.md) docs.
```
Comment (bowenwang1996) [inline docs/develop/node/maintenance/maintenance.md:35]: Same here

File: docs/develop/node/maintenance/maintenance.md [hunk 6]
```diff
@@ -42,9 +30,13 @@ Once `MainNet: Restricted` is live, planned updates to `TestNet` and `MainNet` w
...
 Follow the [Stake Wars](https://near.org/stakewars) if you want to participate in fire-drills and test your playbook for emergency updates.
```
Comment (bowenwang1996) [inline docs/develop/node/maintenance/maintenance.md:36]: Stake wars has ended a while ago. We should probably update this part.

**Review by bowenwang1996 — COMMENTED** [review]
Great work 🚀 🚀 🚀

**Comment by thisisjoshford** [comment]
> Screen Shot 2021-06-07 at 1 06 34 PM
> 
> CI test-link is complaining about broken links on master, not the feature branch ¯\_(ツ)_/¯.

Just migrated to a new CI link checking solution and did not whitelist the 'EDIT' button (instant PR) in the upper right hand corner when creating new pages. Just pushed a fix and will merge it with this branch that should solve these 404 errors.

**Review by janewang — COMMENTED** [review]
@bowenwang1996  @frol @mikedotexe @chefsale @mhalambek @thisisjoshford 

Hi All, I've split the Node Hardware Requirements into Minimal Specifications and Recommended Specifications for Validator, RPC, and Archival. For Minimal, we'd like to keep the requirements as low as reasonably possible. Could you please take a look and comment on these. Thanks!

**Review by bowenwang1996 — APPROVED** [review]

File: docs/develop/node/community/node-community-updates.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+2. Bugs and feature enhancements are tracked in the Github project tracker [`Node Experience`](https://github.com/orgs/near/projects/18). Node operators are welcome to create new issues for features and bugs, and to add these issues into the [Github project tracker](https://github.com/orgs/near/projects/18).
+    - **New feature / Enhancement requests:**  
+        - Please create a [`Github issue`](https://github.com/near/nearcore/issues)
+        - Tag `nodeX`
+        - Filter them through `Incoming Requests` column in the Github project tracker [`Node Experience`](https://github.com/orgs/near/projects/18), where they will be groomed and moved into `Backlog` based on priority
+    - **Bug reports:**
+        - Please create a [`Github issue`](https://github.com/near/nearcore/issues)
+        - Tag `nodeX` and `Bug`
+        - Put it at the top of the `Incoming Requests` column for more immediate attention in the Github project tracker [`Node Experience`](https://github.com/orgs/near/projects/18)
```
Comment (frol) [inline docs/develop/node/community/node-community-updates.md:22]: non-NEAR members cannot label issues or move issues on those boards. We need to create GitHub templates with the pre-set labels to let people create the issues with the specific labels.

File: docs/develop/node/intro/what-is-a-node.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+NEAR Protocol runs on a collection of publicly maintained computers (or "nodes"). All nodes are running the same `nearcore` codebase with the latest releases available on [Github](https://github.com/near/nearcore/releases/). In the following section, we will introduce three types of nodes.
+
+It is important to keep in mind all nodes runs the same codebase, but with different configurations. As such we have split up the instructions for running different nodes into different sections.
+
+## Why run a Node?
+
+You may decide to run a node of your own for a few reasons:
+
+- To join a network as a validator running a "validator node". Running a validator node is a public good and you are effectively securing the NEAR network and earning rewards.
+- To develop and deploy contracts on a node connected to `MainNet`, `TestNet` or `BetaNet` (†)
```
Comment (frol) [inline docs/develop/node/intro/what-is-a-node.md:17]: I don't see it to be a good reason to run your own node. I think it is better to say something along these lines:

> To run applications that heavily depend on RPC performance and/or availability.

File: docs/develop/node/intro/what-is-a-node.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+- To develop and deploy contracts on a local (independent and isolated) node (sometimes called "LocalNet"). (††)
+- To quickly extract blockchain data that can be used for chain analytics
+
+_( † ) `TestNet` is intended to operate as similarly to `MainNet`  as possible with only stable releases while `BetaNet` follows a weekly release cycle._
+
+_( †† ) `LocalNet` would be the right choice if you prefer to avoid leaking information about your work during the development process since `TestNet` and `BetaNet` are *public* networks. `LocalNet` also gives you total control over accounts, economics and other factors for more advanced use cases (ie. making changes to `nearcore`)._
```
Comment (frol) [inline docs/develop/node/intro/what-is-a-node.md:23]: I believe we stick to lowercase naming (`testnet`, `betanet`, `localnet`), please, double-check with @bowenwang1996 and @thisisjoshford

File: docs/develop/node/validator/running-a-node-macos-linux.md [hunk 10]
```diff
@@ -181,10 +168,6 @@ Using local binary at path/to/nearcore/target/release
...
 Stake for user 'stakingpool.youraccount.betanet' with 'ed25519:6ftve9gm5dKL7xnFNbKDNxZXkiYL2cheTQtcEmmLLaW'
 Starting NEAR client...
-Node is running! 
+Node is running!
 To check logs call: `nearup logs` or `nearup logs --follow`
 ```
-
->Got a question?
-<a href="https://stackoverflow.com/questions/tagged/nearprotocol">
-  <h8>Ask it on StackOverflow!</h8></a>
```
Comment (frol) [inline docs/develop/node/validator/running-a-node-macos-linux.md:190]: Is there a better way to get in touch? I believe StackOverflow is the best wiki platform

**Review by frol — COMMENTED** [review]
@janewang Enormous work! :heart_eyes:

**Comment by chefsale** [comment]
@janewang I just wanted to mention that the cost for disk usage seems very very low. Not sure how we got the numbers, but the calculations I made are in hundreds of dollars. We have to take into account that we are using SSDs instead of regular HDDs. We should update that. cc: @frol @bowenwang1996

**Comment by chefsale** [comment]
The cost ranges from 0.10-0.35 cents depending on tyoe of SSDs which are used across providers. For 1.5TB the cost would be somewhere 150-500 dollars depending on setup.

File: docs/develop/node/intro/what-is-a-node.md (hunk 9 above)
Comment (janewang) [reply docs/develop/node/intro/what-is-a-node.md:23]: > I believe we stick to lowercase naming (`testnet`, `betanet`, `localnet`), please, double-check with @bowenwang1996 and @thisisjoshford
Yes, I was thinking of the same naming convention while reading the docs. Sometimes they are TitleCase, sometimes camelCase, and sometimes lowercase. In code, they seem to be consistently lowercase. I will convert all to lowercase for consistency.

File: docs/develop/node/validator/running-a-node-macos-linux.md (hunk 10 above)
Comment (janewang) [reply docs/develop/node/validator/running-a-node-macos-linux.md:190]: > Is there a better way to get in touch? I believe StackOverflow is the best wiki platform

Is the case that most questions for validators are posted on StackOverflow?  Do the core team actively monitor Stackoverflow? 
 
cc @bowenwang1996 @frol @mikedotexe 

Here are some ways recommended to get in touch. Please let me know StackOverflow fits in here too, and I can add back the footer if StackOverflow is actively used:
https://docs.near.org/docs/develop/node/community/node-community-updates

File: docs/develop/node/intro/what-is-a-node.md (hunk 8 above)
Comment (janewang) [reply docs/develop/node/intro/what-is-a-node.md:17]: > The cost ranges from 0.10-0.35 cents depending on tyoe of SSDs which are used across providers. For 1.5TB the cost would be somewhere 150-500 dollars depending on setup.

Yes!
@chefsale  May I please have some guidance and ballpark cost estimation numbers on hardware costs:

Validator node cost estimate: https://docs.near.org/docs/develop/node/validator/hardware
RPC node cost estimate: https://docs.near.org/docs/develop/node/rpc/hardware-rpc
Archival node cost estimate: https://docs.near.org/docs/develop/node/archival/hardware-archival - Is the storage cost of $150-$500 a monthly figure? 

Thanks!

File: docs/develop/node/validator/running-a-node-macos-linux.md (hunk 10 above)
Comment (frol) [reply docs/develop/node/validator/running-a-node-macos-linux.md:190]: It is ideal if people get in touch through StackOverflow. We have the integration that pushes SO questions to Slack, and we keep the questions answered most of the time: https://stackoverflow.com/questions/tagged/nearprotocol

Comment (janewang) [reply docs/develop/node/validator/running-a-node-macos-linux.md:190]: Ooh, neato, I will add it back. Also thanks for commenting on Stackoverflow!

**Pull request merged**